crates/oxc_transformer/src/react/jsx_self/mod.rs

```diff
@@ -29,6 +33,17 @@ impl<'a> ReactJsxSelf<'a> {
     }
 
     pub fn transform_jsx_opening_element(&self, elem: &mut JSXOpeningElement<'a>) {
+        for item in &elem.attributes {
```

Comment:
I think we should avoid this loop on every opening element, this should be checked while transforming.

And I wonder whether we should only emit the diagnostic only once per file, otherwise there's going to be a lot. Can you double check with babel?

Reply:
> And I wonder whether we should only emit the diagnostic only once per file, otherwise there's going to be a lot. Can you double check with babel?

I don't see the logic of babel only reporting once https://github.com/babel/babel/blob/acf3d17fdfe150a270c822581b709dddac4548ce/packages/babel-plugin-transform-react-jsx/src/create-plugin.ts#L529-L531